packages/core/src/modules/oob/OutOfBandApi.ts

```diff
@@ -116,108 +122,121 @@ export class OutOfBandApi {
   }
 
   /**
-   * Creates an outbound out-of-band record containing out-of-band invitation message defined in
-   * Aries RFC 0434: Out-of-Band Protocol 1.1.
+   * Creates an outbound out-of-band record containing out-of-band invitation message of requested version
+   *  - `v1` and default -  Aries RFC 0434: Out-of-Band Protocol 1.1.
+   *  - `v2` -  DIDComm Messaging v2.x: Out-of-Band Protocol 2.0.
    *
-   * It automatically adds all supported handshake protocols by agent to `handshake_protocols`. You
-   * can modify this by setting `handshakeProtocols` in `config` parameter. If you want to create
-   * invitation without handshake, you can set `handshake` to `false`.
+   * For `v1`:
+   *    It automatically adds all supported handshake protocols by agent to `handshake_protocols`. You
+   *    can modify this by setting `handshakeProtocols` in `config` parameter. If you want to create
+   *    invitation without handshake, you can set `handshake` to `false`.
    *
-   * If `config` parameter contains `messages` it adds them to `requests~attach` attribute.
+   *    If `config` parameter contains `messages` it adds them to `requests~attach` attribute.
    *
-   * Agent role: sender (inviter)
+   *    Agent role: sender (inviter)
    *
    * @param config configuration of how out-of-band invitation should be created
    * @returns out-of-band record
    */
   public async createInvitation(config: CreateOutOfBandInvitationConfig = {}): Promise<OutOfBandRecord> {
-    const multiUseInvitation = config.multiUseInvitation ?? false
-    const handshake = config.handshake ?? true
-    const customHandshakeProtocols = config.handshakeProtocols
-    const autoAcceptConnection = config.autoAcceptConnection ?? this.connectionsApi.config.autoAcceptConnections
-    // We don't want to treat an empty array as messages being provided
-    const messages = config.messages && config.messages.length > 0 ? config.messages : undefined
-    const label = config.label ?? this.agentContext.config.label
-    const imageUrl = config.imageUrl ?? this.agentContext.config.connectionImageUrl
-    const appendedAttachments =
-      config.appendedAttachments && config.appendedAttachments.length > 0 ? config.appendedAttachments : undefined
-
-    if (!handshake && !messages) {
-      throw new AriesFrameworkError(
-        'One or both of handshake_protocols and requests~attach MUST be included in the message.'
-      )
-    }
+    let outOfBandRecord: OutOfBandRecord | null
+    if (config.version === 'v2') {
+      const outOfBandInvitation = await this.v2OutOfBandService.createInvitation(this.agentContext)
+      outOfBandRecord = new OutOfBandRecord({
+        role: OutOfBandRole.Sender,
+        state: OutOfBandState.AwaitResponse,
+        alias: config.alias,
+        v2OutOfBandInvitation: outOfBandInvitation,
```

Comment:
Hi @TimoGlastra I am looking into doing this, would you mind to elaborate a little further?

Reply:
If you look at e.g. the CredentialExchangeRecord, it doesn't store any of the didcomm messages. Rather, they are store in a DidCommMessageRecord, and fetched/stored when received/created.

I was thinking that we can maybe do the same for the out of band record, meaning we remove the outOfBandInvitation form the OutOfBandRecord, and instead store it separately in a DidCommMessageRecord. The code for it would look something like:

```
    await didCommMessageRepository.saveAgentMessage(agentContext, {
      agentMessage: outOfBandInvitation,
      role: DidCommMessageRole.Sender, // or .Receiver
      associatedRecordId: outOfBandRecord.id,
    })

       const outOfBandInvitation = await this.didCommMessageRepository.getAgentMessage(messageContext.agentContext, {
      associatedRecordId: outOfBandRecord.id,
      messageClass: V1OutOfBandInvitation,
    })
```

We would need to update the DidCommMessageRecord to also support DIDComm v2

Reply:
Thanks, I will try to do that. :)